indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -844,6 +858,48 @@ private void removeWorker(final Worker worker)
     }
   }
 
+  private void scheduleTasksCleanupForWorker(final String worker, final List<String> tasksToFail)
+  {
+    removedWorkerCleanups.put(
+        worker, exec.schedule(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                log.info("Running scheduled cleanup for Worker[%s]", worker);
+                for (String assignedTask : tasksToFail) {
```

Comment:
the cleanup gets cancelled in addWorker() if the worker shows up within grace period, So in a spotty network the tasks will not fail.